lib/kino_flame/runner_cell.ex

```diff
@@ -0,0 +1,114 @@
+defmodule KinoFLAME.RunnerCell do
+  @moduledoc false
+
+  use Kino.JS, assets_path: "lib/assets/runner_cell", entrypoint: "build/main.js"
+  use Kino.JS.Live
+  use Kino.SmartCell, name: "FLAME runner cell"
+
+  @text_fields ["name"]
+  @number_fields ["min", "max", "max_concurrency"] ++ ["fly_cpus", "fly_memory_gb", "fly_gpus"]
+
+  @impl true
+  def init(attrs, ctx) do
+    fields = %{
+      "name" => Kino.SmartCell.prefixed_var_name("runner", attrs["name"]),
+      "min" => attrs["min"] || 0,
+      "max" => attrs["max"] || 1,
+      "max_concurrency" => attrs["max_concurrency"] || 10,
+      "fly_cpus" => attrs["fly_cpus"] || 1,
+      "fly_cpu_kind" => attrs["fly_cpu_kind"] || "shared",
+      "fly_memory_gb" => attrs["fly_memory_gb"] || 1,
+      "fly_gpu_kind" => attrs["fly_gpu_kind"],
+      "fly_gpus" => attrs["fly_gpus"]
+    }
+
+    {:ok, assign(ctx, fields: fields)}
+  end
+
+  @impl true
+  def handle_connect(ctx) do
+    payload = %{
+      fields: ctx.assigns.fields
+    }
+
+    {:ok, payload, ctx}
+  end
+
+  @impl true
+  def handle_event("update_field", %{"field" => field, "value" => value}, ctx) do
+    updated_fields = to_updates(field, value)
+    ctx = update(ctx, :fields, &Map.merge(&1, updated_fields))
+    broadcast_event(ctx, "update", %{"fields" => updated_fields})
+    {:noreply, ctx}
+  end
+
+  defp to_updates(field, value) when field in @number_fields and is_binary(value) do
+    value =
+      case Integer.parse(value) do
+        {n, ""} -> n
+        _ -> nil
+      end
+
+    %{field => value}
+  end
+
+  defp to_updates(field, "") when field not in @text_fields, do: %{field => nil}
+
+  defp to_updates(field, value), do: %{field => value}
+
+  @impl true
+  def to_attrs(ctx) do
+    ctx.assigns.fields
+  end
+
+  @impl true
+  def to_source(attrs) do
+    required_keys =
+      ["name", "min", "max", "max_concurrency"] ++
+        ["fly_cpu_kind", "fly_cpus", "fly_memory_gb"]
+
+    if all_fields_filled?(attrs, required_keys) do
+      attrs |> to_quoted() |> Kino.SmartCell.quoted_to_string()
+    else
+      ""
+    end
+  end
+
+  defp all_fields_filled?(attrs, keys) do
+    not Enum.any?(keys, fn key -> attrs[key] in [nil, ""] end)
+  end
+
+  defp to_quoted(attrs) do
+    specs_opts =
+      [
+        cpu_kind: attrs["fly_cpu_kind"],
+        cpus: attrs["fly_cpus"],
+        memory_mb: attrs["fly_memory_gb"] * 1024,
+        gpu_kind: attrs["fly_gpu_kind"],
+        gpus: attrs["fly_gpus"]
+      ]
+      |> Enum.reject(&(elem(&1, 1) == nil))
+
+    # TODO try changing FLAME to use /.fly/api instead of :token
```

Comment:
Just to confirm, FLAME is going to automatically pick the same Docker image as the one running Livebook, right?